### DISCUSSION TITLE
Editing resource that has a HasOne self reference results in error

### DISCUSSION THREAD

**Discussion by sedlatschek** [laravel/nova-issues#5435] [Q&A] [answered]
- Laravel Version: 10.4.1
- Nova Version: 4.22.2
- PHP Version: 8.1.17
- Database Driver & Version: SQLite 
- Operating System and Version: Win 11
- Reproduction Repository: https://github.com/sedlatschek/laravel-10-nova-4-issues

### Description:

Hi,

I have a model called `Deadline`, which has two self-referencing relations called `postponed` and `postpones`:

```php
class Deadline extends Model
{
    use HasFactory;

    protected $fillable = [
        'exceeds_at',
        'postponed_id',
    ];

    protected $casts = [
        'exceeds_at' => 'datetime',
    ];

    public function postponed(): BelongsTo
    {
        return $this->belongsTo(Deadline::class);
    }

    public function postpones(): HasOne
    {
        return $this->hasOne(Deadline::class, 'postponed_id');
    }
}
```

The nova resource for this model looks like this:

```php
class Deadline extends Resource
{
    public static $model = \App\Models\Deadline::class;

    public function fields(NovaRequest $request)
    {
        return [
            ID::make()->sortable(),
            Date::make('exceeds_at'),
            BelongsTo::make('Postponed To', 'postponed', Deadline::class)
                ->readonly(),
            HasOne::make('Postponed From', 'postpones', Deadline::class)
                ->readonly(),
        ];
    }
}
```

I can view the resources without any problems:
![image](https://user-images.githubusercontent.com/58665536/227927456-38bacdb8-31ec-4fdd-8b06-4d824f6d53f9.png)

But when I click on the edit button, I get endless loading:
![image](https://user-images.githubusercontent.com/58665536/227927676-42bf4812-f5cd-4d01-847d-8bdf1673c9be.png)

The nova-api request returns HTTP 500 after either running out of memory or hitting the maximum execution time.

The Update.vue then runs into the following error:
```
Update.vue:165 Uncaught (in promise) TypeError: Cannot destructure property 'data' of '(intermediate value)' as it is undefined.
    at Proxy.getFields (Update.vue:165:9)
```

### Detailed steps to reproduce the issue on a fresh Nova installation:

1. Clone https://github.com/sedlatschek/laravel-10-nova-4-issues
2. `php artisan migrate:fresh --seed`
3. Open any `Deadline` resource
4. Click on the edit button

Thanks for your help

**Comment by crynobone** [comment, marked as answer]
It might be best not to include inline `HasOne` on create/edit for this structure. e.g: 

```php
HasOne::make('Postponed From', 'postpones', Deadline::class)->exceptOnForms(),

**Comment by sedlatschek** [comment]
Thanks for your answer.

What is the desired behaviour for this case? HasOne relations that are no self-reference do not require me to add `exceptOnForms()` to work.
